package.json

```diff
@@ -6,7 +6,8 @@
     "lint": "eslint . --ext .ts --fix",
     "build": "tsc",
     "dev": "nodemon src/index.ts",
-    "start": "node build/src/index.js"
+    "start": "node build/src/index.js",
+    "test": "yarn jest --forceExit"
```

Comment:
Any reason to use --forceExit herer?

Reply:
Yes, forgot to mention in the comment - [redis](https://github.com/snapshot-labs/blockfinder/blob/master/src/redis.ts) is still running after the tests has gone through. Another option would be to kill the process in the test file, unless you have another suggestion?

Reply:
Ok I see, I think for the tests we shouldn't use redis, we should adapt the function so we can disable cache if we need. Also we should support redis caching for the function to get block from the API too. Maybe caching should happen on the `tsToBlockNum` function level and not in `tsToBlockWithApi` or `tsToBlockWithNode`. This will also reduce the number of request to the API.

Reply:
Sorry this comment is misleading, actually Redis is setup in a parent function. Maybe we can move tsToBlockWithApi and tsToBlockWithNode to another file so it doesnt import redis to run the tests.

Reply:
So caching is actually happening in the `query` function as it was [before](https://github.com/snapshot-labs/blockfinder/blob/master/src/graphql/blocks.ts#L86-L91), so once cached whether through API or Node, there are no further calls. Unless I'm missing something?

I'll try to figure out how to disable it for testing :pray:

Reply:
Mocking redis solved it!

`jest.mock('../../src/redis.ts', () => jest.fn());`